src/csharp/Microsoft.Spark/Hadoop/FS/FileSystem.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Spark.Hadoop.Conf;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Hadoop.FS
+{
+    /// <summary>
+    /// A fairly generic filesystem. It may be implemented as a distributed filesystem, or as a "local" one
+    /// that reflects the locally-connected disk. The local version exists for small Hadoop instances and for
+    /// testing.
+    /// 
+    /// All user code that may potentially use the Hadoop Distributed File System should be written to use a
+    /// FileSystem object. The Hadoop DFS is a multi-machine system that appears as a single disk. It's
+    /// useful because of its fault tolerance and potentially very large capacity.
+    /// </summary>
+    public class FileSystem : IJvmObjectReferenceProvider, IDisposable
+    {
+        private readonly JvmObjectReference _jvmObject;
+
+        internal FileSystem(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Returns the configured <see cref="FileSystem"/>.
+        /// </summary>
+        /// <param name="conf">The configuration to use.</param>
+        /// <returns>The FileSystem.</returns>
+        public static FileSystem Get(Configuration conf) =>
+            new FileSystem((JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                "org.apache.hadoop.fs.FileSystem", "get", conf));
```

Comment:
nit: I believe you need to break for each param.

Reply:
Fixed